src/bptl/work_units/zgw/tasks/zaak.py

```diff
@@ -231,3 +232,61 @@ def perform(self):
             "archiefnominatie": resultaat["archiefnominatie"],
             "archiefactiedatum": resultaat["archiefactiedatum"],
         }
+
+
+@register
+class LookupZaak(ZGWWorkUnit):
+    """
+    Look up a single ZAAK by identificatie and bronorganisatie.
+
+    This task looks up the referenced zaak, and if found sets the zaakUrl as a
+    process variable. If not found, the variable will be empty.
+
+    You can use this to check if the referenced ZAAK does indeed exist, and relate
+    it to other objects.
+
+    **Required process variables**
+
+    * ``identificatie``: identification of the zaak, commonly known as "zaaknummer"
+    * ``bronorganisatie``: RSIN of the source organization for the zaak. The combination
+        of identificatie and bronorganisatie uniquely identifies a zaak.
+    * ``services``: JSON Object of connection details for ZGW services:
+
+        .. code-block:: json
+
+          {
+              "<zrc alias>": {"jwt": "Bearer <JWT value>"},
+          }
+
+    **Optional process variables (Camunda exclusive)**
+
+    * ``callbackUrl``: send an empty POST request to this URL to signal completion
+
+    **Sets the process variables**
+
+    * ``zaakUrl``: the URL reference of the retrieved zaak, if retrieved at all. If the
+        zaak was not found, the value will be ``null``
+    """
+
+    def perform(self) -> Dict[str, Optional[str]]:
+        variables = self.task.get_variables()
+        client_zrc = self.get_client(APITypes.zrc)
+
+        identificatie = check_variable(variables, "identificatie")
+        bronorganisatie = check_variable(variables, "bronorganisatie")
+
+        zaken: dict = client_zrc.list(
+            "zaak",
+            {
+                "bronorganisatie": bronorganisatie,
+                "identificatie": identificatie,
+            },
+        )
+
+        # paginated response
+        if zaken["results"]:
+            zaak_url = zaken["results"][0]["url"]
+        else:
+            zaak_url = None
```

Comment:
Should None not be an empty string since zaak_url is a string normally?

Reply:
Discussed on Slack, we both can't give a hard reason why it should be one or the other.

Checked with Java's interpretation/meaning of `null` and it matches the intended meaning, i.e. the object does not exist. However, changing types in Python is kinda iffy. We'll see if we run into problems or not - maybe down the road with the Camunda Expression Language there will be a clear reason to pick one or the other.